gateway/gateway.go

```diff
@@ -7,24 +7,52 @@ import (
 )
 
 type Gateway struct {
-	distributorProxy *Proxy
-	server           *server.Server
+	DistributorProxy   *Proxy
```

Comment:
```suggestion
	distributorProxy   *Proxy
```
Let's keep this local